### PR TITLE
Add APP_ENV flag for envs

### PR DESCRIPTION
**Pull request by jtmst**
## Description
Relates to #[14977](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14977). (or closes?)

## Testing done
Local testing with mock env files

## Screenshots
Screen Shot 2023-09-22 at 12 13 10 PM
Screen Shot 2023-09-22 at 12 15 43 PM

## QA steps
- Run `yarn dev`
  - Verify in network tab that static-entry files are coming from the local generated files
- Create a .env.staging file with a value of  NEXT_PUBLIC_ASSETS_URL=https://s3-us-gov-west-1.amazonaws.com/content.staging.va.gov/generated/ 
  - run `APP_ENV=staging yarn dev'
  - verify in network tab that static-entry files are coming from the appropriate staging bucket

Note:
- This adds the capability to call on different env files with the APP_ENV flag, but still will require that those env files are created in their appropriate environments

## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -61,6 +61,15 @@
       "rules": {
         "import/no-anonymous-default-export": "off"
       }
+    },
```
Comment (tjheffner) [inline .eslintrc.json:64]: per slack thread with the 3 of us, this can be removed

File: README.md [hunk 2]
```diff
@@ -37,6 +37,16 @@ You should set these up before attempting to install the repo.
 You will now have a Next.js development server running at http://localhost:3000, which will refresh with changes to your
 local environment.
 
+### Environment Flags
+
+The APP_ENV flag can be used to designate which .env file you want to use. Ensure you're env files are in ./env, and then run your build or dev command with a leading APP_ENV=local.
```
Comment (tjheffner) [inline README.md:42]: I see you moved `.env.example` but not `.env.test`, just curious if `.env.test` gets picked up when it lives in this folder? We preface a bunch of package.json's `test:` scripts with `NODE_ENV=test`

**Review by tjheffner — CHANGES_REQUESTED** [review]
Per https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables next supports and has special handling for `.env.local`, `.env.development`, `.env.production` and `.env.test` in the root directory out of the box... I am not opposed to not using the root level and moving them to `./envs/`, and your changes seem to pick up the exposed `NEXT_PUBLIC_` variables, but please double check about `.env.test` and its location.


the changes to yarn.lock again seem huge for what is ultimately being added

**Comment by jtmst** [comment]
> Per https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables next supports and has special handling for `.env.local`, `.env.development`, `.env.production` and `.env.test` in the root directory out of the box... I am not opposed to not using the root level and moving them to `./envs/`, and your changes seem to pick up the exposed `NEXT_PUBLIC_` variables, but please double check about `.env.test` and its location.
> 
> the changes to yarn.lock again seem huge for what is ultimately being added

I'll adjust for the test environments and push that up.  

Nexts built in env support forces you to use the local env file when running `yarn dev`, and requires some extra set up to support envs besides those listed (staging).  I think this is the approach we want, once i fix this for test.

**Review by tjheffner — APPROVED** [review]

**Pull request merged**